pkcs11/pkcs11_test.go

```diff
@@ -594,3 +595,77 @@ func TestCreateCertificate(t *testing.T) {
 		t.Errorf("Returned certificate did not match loaded certificate")
 	}
 }
+
+func TestDecrypt_OAEP(t *testing.T) {
+	msg := "Plain text to encrypt"
+	b := []byte(msg)
+	tests := []struct {
+		name string
+		bits int
+	}{
+		{"2048", 2048},
+		{"4096", 4096},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			s := newTestSlot(t)
+			o := keyOptions{RSABits: test.bits}
+			priv, err := s.generate(o)
+			if err != nil {
+				t.Fatalf("generate(%#v) failed: %v", o, err)
+			}
+			rsaPub := priv.(*rsaPrivateKey).pub
+			// SHA1 is the only hash function supported by softhsm
+			cipher, err := rsa.EncryptOAEP(sha1.New(), rand.Reader, rsaPub, b, nil)
```

Comment:
nit: use SHA 256 instead of SHA 1. The latter is broken and you generally need to get allow listed internally to use it. Easier to use a non-broken hash.

Reply:
Would rather have the test and go through the allow list internally. That's fine!